spec/system/custom/polls/polls_spec.rb

```diff
@@ -33,29 +49,195 @@
       login_as user
       visit poll_path(poll)
 
-      within("#poll_question_#{question.id}_answers") do
-        expect(page).to have_link("Yes")
-        expect(page).to have_link("No")
+      expect(page).to have_checked_field("No")
+    end
+
+    scenario "Question answers appear in the given order" do
+      question = create(:poll_question, poll: poll)
+      answer1 = create(:poll_question_answer, title: "First answer", question: question, given_order: 2)
+      answer2 = create(:poll_question_answer, title: "Second answer", question: question, given_order: 1)
+
+      visit poll_path(poll)
+
+      expect(answer2.title).to appear_before(answer1.title)
+    end
+
+    scenario "Level 2 users in an expired poll" do
+      expired_poll = create(:poll, :expired, geozone_restricted: true)
+      expired_poll.geozones << geozone
+
+      create(:poll_question, :yes_no, poll: expired_poll)
+
+      login_as(create(:user, :level_two, geozone: geozone))
+
+      visit poll_path(expired_poll)
+
+      expect(page).to have_field("Yes", disabled: true)
+      expect(page).to have_field("No", disabled: true)
+      expect(page).to have_content("This poll has finished")
+    end
+
+    scenario "Show form errors when any answer is invalid" do
+      poll = create(:poll)
+      open_question = create(:poll_question, poll: poll, mandatory_answer: true)
+      single_choice_question = create(:poll_question, :yes_no, poll: poll, mandatory_answer: true)
+      login_as(create(:user, :verified))
+
+      visit poll_path(poll)
+      click_button "Send"
+
+      within "#question_#{open_question.id}_answer_fields" do
+        expect(page).to have_content("can't be blank")
+      end
+      within "#question_#{single_choice_question.id}_answer_fields" do
+        expect(page).to have_content("Answer can't be blank")
+      end
+    end
+
+    scenario "Show fullfilled form when answers were saved successfully" do
+      Setting["feature.user.skip_verification"] = true
```

Comment:
Why are we skipping the verification? Is it related to the fact that we're using `login_as` in the tests even though we expect people not to log in before voting? If so, perhaps we should remove the `login_as` in these custom tests :thinking:.

Reply:
We need to skip verification always because of a [validation rule we have](https://github.com/rockandror/innoviris-consul/blob/stable/app/models/poll/voter.rb#L17) at the `Poll::Voter` model that validates the presence of the document_number when this feature is not enabled.

Reply:
:+1: